tests/Jira/ApiTest.php

```diff
@@ -14,116 +15,164 @@
 class ApiTest extends \PHPUnit_Framework_TestCase
 {
 
+	const ENDPOINT = 'http://jira.company.com';
+
+	/**
+	 * Api.
+	 *
+	 * @var Api
+	 */
+	protected $api;
+
 	/**
-	 * Tests that any trailing slash in the endpoint url is removed before being stored in the object state
+	 * Credential.
+	 *
+	 * @var AuthenticationInterface
 	 */
-	public function testSetEndpointTrailingSlash()
+	protected $credential;
+
+	/**
+	 * Client.
+	 *
+	 * @var ObjectProphecy
+	 */
+	protected $client;
+
+	protected function setUp()
 	{
-		$api = new Api('https://test.test/', new Anonymous(), null);
-		$this->assertEquals('https://test.test', $api->getEndpoint());
+		parent::setUp();
 
-		// Make sure nothing is removed if there is no trailing slash
-		$url = 'https://urlwithouttrailing.slash';
-		$api->setEndPoint($url);
-		$this->assertEquals($url, $api->getEndpoint());
+		$this->credential = $this->prophesize('chobie\Jira\Api\Authentication\AuthenticationInterface')->reveal();
+		$this->client = $this->prophesize('chobie\Jira\Api\Client\ClientInterface');
+
+		$this->api = new Api(self::ENDPOINT, $this->credential, $this->client->reveal());
 	}
 
 	/**
-	 * Tests that the updateVersion call constructs the correct api call
+	 * @dataProvider setEndpointDataProvider
 	 */
+	public function testSetEndpoint($given_endpoint, $used_endpoint)
+	{
+		$api = new Api($given_endpoint, $this->credential, $this->client->reveal());
+		$this->assertEquals($used_endpoint, $api->getEndpoint());
+	}
+
+	public function setEndpointDataProvider()
+	{
+		return array(
+			'trailing slash removed' => array('https://test.test/', 'https://test.test'),
+			'nothing removed' => array('https://test.test', 'https://test.test'),
+		);
+	}
+
 	public function testUpdateVersion()
 	{
 		$params = array(
-			'released' => true,
-			'releaseDate' => '2010-07-06',
+			'overdue' => true,
```

Comment:
Looking at previous test code it was obvious that sample data was copy-pasted from `releaseVersion` test, where exactly that data is submitted to JIRA.

In there we testing version update and that have nothing to do with indirect version release (it can be used to release if user wishes so, but that's not primary purpose of this method). To point that out even more I've used completely unrelated to releasing fields of the version.